homework_2/library.py

```diff
@@ -0,0 +1,60 @@
+import json
+from typing import List
+from abc import ABC, abstractmethod
+
+from models import BookModel, MagazineModel
+from decorators import log_adding_publication, is_publication_exist
+
+
+class AbstractLibrary(ABC):
+    @abstractmethod
+    def add_publication(self, publication: BookModel):
+        raise NotImplementedError
+
+    @abstractmethod
+    def remove_publication(self, publication: BookModel):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_publication_by_author(self, author: str):
+        raise NotImplementedError
+
+
+class Library:
+    def __init__(self):
+        self.collection: List[BookModel] = []
+
+    def get_publication_list(self) -> list[BookModel]:
+        for publication in self.collection:
+            print(publication)
+
+    @log_adding_publication
+    def add_publication(self, publication: BookModel):
+        self.collection.append(publication)
+
+    @is_publication_exist
+    def remove_publication(self, publication: BookModel):
+        if publication in self.collection:
+            self.collection.remove(publication)
+
+    def get_publication_by_author(self, author: str):
+        for publication in self.collection:
+            if publication.author == author:
+                print(publication)
```

Comment:
I'd suggest to better create an empty list and fill it with filtered publications and then print it

Reply:
Thanks, fixed